drivers/cache/Kconfig.stm32

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2025 Henrik Lindblom <henrik.lindblom@vaisala.com>
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig CACHE_STM32
+	bool "STM32 cache driver"
+	select CACHE_HAS_DRIVER
+	depends on CACHE_MANAGEMENT
+	help
+	  Enable support for the STM32 ICACHE / DCACHE peripheral present in some STM32 chips.
+
+if CACHE_STM32
+
+# "default n" for L5 is legacy - could be removed?
+config CACHE_STM32_ICACHE_DIRECT_MAPPING
+	bool "Use 1-way associative mapping for ICACHE"
+	default n if SOC_SERIES_STM32L5X
```

Comment:
I think it should be ok to move L5 to 1-way associated mapping, but we should document the change in migration guide and mention how to revert.